lib/cadet/devices/devices.ex

```diff
@@ -0,0 +1,348 @@
+defmodule Cadet.Devices do
+  @moduledoc """
+  Contains domain logic for remote execution devices.
+  """
+  use Cadet, :context
+
+  import Ecto.Query
+
+  alias Cadet.AwsHelper
+  alias Cadet.Accounts.User
+  alias Cadet.Devices.{Device, DeviceRegistration}
+  alias ExAws.STS
+
+  @spec get_user_registrations(integer | %User{}) :: [%DeviceRegistration{}]
+  def get_user_registrations(%User{id: user_id}) do
+    get_user_registrations(user_id)
+  end
+
+  def get_user_registrations(user_id) when is_integer(user_id) or is_binary(user_id) do
```

Comment:
Seems a bit peculiar `get_user_registrations/1` here accepts a binary `user_id` whilst `get_user_registration/2` and `register/4` below does not.